### ISSUE TITLE
Exec or equivalent through environment config

### ISSUE THREAD

**Issue by dn5** [andres-montanez/Magallanes#336]
Since we now have `filesystem` functions through configuration, I'd like to hear an input regarding the equivalent for `exec(ution)`. A lot of enterprise or large-scale projects have for example a binary applications that does some things quite necessary for a specific environment. Currently, this is at best dealt through custom task. 

I'd prefer a choice through config (`.mage.yml`) file as:

```
magephp:
    environments:
        production:
            post-deploy:
                - exec: { 'Return jobs status', './bin/supervisorctl status all > deploy_status.txt' }
                - exec: { 'Restart Supervisor jobs', './bin/supervisorctl restart all' }
```

My design would prefer a description of the execution command, followed by the second argument - the actual execution.

This of course should be defined by collective voting here, on whats the prettiest syntax to use. The following command could then be extended to various different settings and functionalities.

**Comment by andres-montanez** [comment]
HI @dn5 it's an interesting idea! I will take a look at it.

**Comment by dn5** [comment]
Glad to hear! I hope to get a spare time and possibly send a pull request. Looking forward for new versions, you are doing Gods work :)

**Comment by dn5** [comment]
Note: This is low priority task so merge can wait for further version.

**Comment by Toflar** [comment]
@dn5 enjoy: https://github.com/andres-montanez/Magallanes/pull/356 ;)

**Comment by dn5** [comment]
Nice. I'd like to see your PR merged in following versions.

**Comment by andres-montanez** [comment]
Hi everyone, @Toflar PR has been merged and the feature will be available in v3.1.
Thanks everyone!
